shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -195,6 +195,18 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
       JSRecord(fields map { case (key, Fld(_, _, _, value)) =>
         key.name -> translateTerm(value)
       })
+    case Sel(Var(cls), Var(u: "unapply")) =>
```

Comment:
Why? Why wouldn't the access to `unapply` be uniform across different kinds of classes? Users may want to call that method explicitly. It should always just be `ClassName.unapply`.

Reply:
By the way this code is crazy. You're messing up with any static class field incidentally named `unapply` here.

Reply:
To elaborate on my suggestion: just use a field in the `ClassName` value, be it a class or a function.

Is there any advantage to using an actual static method here?

Reply:
If we treat `unapply` as a non-static method, we need to consider how to handle the overriding of it.

Reply:
I think you misunderstand my suggestion. How would you override a method that's called as MyClass.foo

Reply:
Removed the unnecessary branch for `unapply`